src/App.tsx

```diff
@@ -1,73 +1,65 @@
-import React from 'react';
+import React, { useMemo, useRef, useState } from 'react';
+import { Autocomplete } from './components/Autocomplete';
 import './App.scss';
 import { peopleFromServer } from './data/people';
+import { Person } from './types/Person';
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [selectedPerson, setSelectedPerson] = useState<Person | null>(null);
+  const { name, born, died } = selectedPerson || {};
```

Comment:
Destructuring `selectedPerson` without checking if it is not null can lead to `name`, `born`, and `died` being undefined. Consider using optional chaining or a conditional check to ensure these values are accessed safely.